includes/discovery/discovery-protocols.inc.php

```diff
@@ -49,19 +50,19 @@
             $interface = get_port_by_ifIndex($device['device_id'], $key);
             d_echo($cdp_if_array);
             foreach ($cdp_if_array as $entry_key => $cdp) {
-                if (is_valid_hostname($cdp['cdpCacheDeviceId']) || ($config['discovery_by_ip'] == true)) {
+                if (is_valid_hostname($cdp['cdpCacheDeviceId']) || Config::get('discovery_by_ip')) {
                     $cdp_ip = IP::fromHexString($cdp['cdpCacheAddress'], true);
                     $remote_device_id = dbFetchCell('SELECT `device_id` FROM `devices` WHERE `sysName` = ? OR `hostname` = ? OR `hostname` = ?', array($cdp['cdpCacheDeviceId'], $cdp['cdpCacheDeviceId'], $cdp_ip));
 
                     if (!$remote_device_id &&
-                        !can_skip_discovery($config['autodiscovery']['cdp_exclude']['platform_regexp'], $cdp['cdpCachePlatform'], $cdp['cdpCacheDeviceId']) &&
-                        !can_skip_discovery($config['autodiscovery']['xdp_exclude']['sysname_regexp'], $cdp['cdpCacheDeviceId'], $cdp['cdpCacheDeviceId']) &&
-                        !can_skip_discovery($config['autodiscovery']['xdp_exclude']['sysdesc_regexp'], $cdp['cdpCacheVersion'], $cdp['cdpCacheDeviceId'])
+                        !can_skip_discovery(Config::get('autodiscovery.cdp_exclude.platform_regexp'), $cdp['cdpCachePlatform'], $cdp['cdpCacheDeviceId']) &&
+                        !can_skip_discovery(Config::get('autodiscovery.xdp_exclude.sysname_regexp'), $cdp['cdpCacheDeviceId'], $cdp['cdpCacheDeviceId']) &&
+                        !can_skip_discovery(Config::get('autodiscovery.xdp_exclude.sysdesc_regexp'), $cdp['cdpCacheVersion'], $cdp['cdpCacheDeviceId'])
                     ) {
-                        if ($config['discovery_by_ip'] !== true) {
-                            $remote_device_id = discover_new_device($cdp['cdpCacheDeviceId'], $device, 'CDP', $interface);
-                        } else {
+                        if (Config::get('discovery_by_ip')) {
```

Comment:
This was a !== check before, is this right?

Reply:
@laf yeah, I flipped the if/else to avoid !Config